[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

ghillairet opened the pull request:
The problem seemed to be that paths were never recoginzed as keys (which they should have been). That's why paths always had a different color that other keys. I was still seeing the last closing bracket before a colon to have a slightly different color than the rest of the path. Adding a special rule to handle path, based on the KeyRule but with a modified regex to account for brackets and slashes seems to fix the problem. The PathRule will highligh paths as scalar so that paths have different color than other keys.

What I am seeing in latest build:

![screen shot 2018-09-05 at 12 07 12](https://user-images.githubusercontent.com/148045/45087137-36f85f80-b105-11e8-9611-6e58d1277176.png)

What I am seeing with this PR fix:

![screen shot 2018-09-05 at 12 05 41](https://user-images.githubusercontent.com/148045/45087194-58594b80-b105-11e8-929e-41ce885ddb6f.png)


What I am seeing if paths have same color than keys:

![screen shot 2018-09-05 at 12 05 27](https://user-images.githubusercontent.com/148045/45087202-5d1dff80-b105-11e8-99ed-c294cdf4e768.png)

ghillairet committed:
[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

The problem seemed to be that paths were never recoginzed as keys (which they should have been). That's why paths always had a different color that other keys. I was still seeing the last closing bracket before a colon to have a slightly different color than the rest of the path. Adding a special rule to handle path, based on the KeyRule but with a modified regex to account for brackets and slashes seems to fix the problem. The PathRule will highligh paths as scalar so that paths have different color than other keys.

tfesenko commented on com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/editor/JsonScanner.java:
```diff
@@ -39,6 +39,8 @@
+import com.reprezen.swagedit.core.editor.scanner.PathRule;
+
 /*
  * Identical implementation of a BufferedRuleBasedScanner than YAMLScanner but makes 
```
With the new `PathRule` this comment is no longer true, it needs to be updated

ghillairet committed:
[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

Make PathRule use it's own token.

ghillairet committed:
[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

Adding a preference constant for path color and adding it to the theme mapping file.
Also removing duplicate code in preference initializer and missing super call in OpenApiColorPreferences.

ghillairet committed:
[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

Extend KeyRule to allow dollar sign and slash to handle references and media types as keys.
Modify PathRule regex to only match string starting with /.
Change Path color to use the same theme color as other keys.

tfesenko commented on com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/editor/JsonScanner.java:
```diff
@@ -68,7 +67,7 @@ protected void init() {
                 PreferenceConstants.ITALIC_KEY, PreferenceConstants.UNDERLINE_KEY);
         IToken keyToken = new YAMLToken(keyAttr, YAMLToken.KEY);
 
-        TextAttribute pathKeyAttr = tokenAttribute(KaizenPreferenceConstants.COLOR_PATH, PreferenceConstants.BOLD_KEY,
```
So, now `KaizenPreferenceConstants.COLOR_PATH` is set in PreferencesInitializer, but it's never used, is that right? It's confusing without knowing the history of changes. I suggest removing it or, if we want to keep it for future, add a comment that it was created to be used for `pathKeyAttr ` in `JsonScanner` and a link to the discussion in PR.

tfesenko reviewed (approved):
Passed code review and QA.
Please see my suggestion about a comment for the unused constant.

ghillairet committed:
[ZEN-4360] OAS3 Syntax Coloring: some characters colored black

Remove unused color path preference.
